src/utils.c

```diff
@@ -645,12 +663,30 @@ SEXP r_maybe_duplicate(SEXP x) {
   }
 }
 
+bool r_is_names(SEXP names) {
+  if (names == R_NilValue) {
+    return false;
+  }
+
+  R_len_t n = Rf_length(names);
+  const SEXP* p = STRING_PTR_RO(names);
+
+  for (R_len_t i = 0; i < n; ++i, ++p) {
+    SEXP nm = *p;
+    if (nm == strings_empty || nm == NA_STRING) {
```

Comment:
This seems a bit too strict a test. I don't think you need to worry about empty/missing names here (and it might occasionally be useful).